src/webpage-digester.ts

```diff
@@ -0,0 +1,135 @@
+import {IndexFilter} from "./filters/index-filter";
+import {Filter} from "./filters/filter";
+import {WebPage} from "./utils/webpage";
+import {Occurrence} from "./utils/occurrence";
+
+
+export class WebPageDigester {
+    private searchTerms : Array<string>;
+    private mainFilter : new () => IndexFilter;
+    private mainFilterInstance : IndexFilter;
+    private preFilterInstance : Filter;
+
+    constructor(searchTerms : Array<string>) {
+        this.searchTerms = searchTerms;
+    }
+
+    /**
+     * Set the main filter of the digester which will determine the indexes of all matches
+     * Be careful which filter to use here as not all filters will give you exact matches!
+     * @param filterConstructor                             Class of the filter
+     */
+    public setFilter<T extends IndexFilter> (filterConstructor : new () => T) : WebPageDigester {
+        this.mainFilter = filterConstructor;
+        this.mainFilterInstance = undefined;
+        return this;
+    }
+
+    /**
+     * Set a pre-filter which will reduce the number of searchTerms for the main filter
+     * If the pre-filter doesn't find any matches it will skip the main filter entirely
+     * This is very useful if your main filter is slow
+     * @param filterConstructor                             Class of the pre-filter
+     */
+    public setPreFilter<T extends Filter> (filterConstructor : new () => T) : WebPageDigester{
+        this.preFilterInstance = new filterConstructor();
+        this.preFilterInstance.addSearchTerms(this.searchTerms);
+        return this;
+    }
+
+    public removePreFilter() : WebPageDigester {
```

Comment:
It only removes the pre-filter. You currently can only have a singe pre-filter. I couldn't think of a use case for multiple pre-filters.